### PR TITLE
Fixing problem with saving birthday

### PR DESCRIPTION
**Pull request by fism88**
Should be able to save birthdays for different users. #163

Added cancel button to each of the sections in the edit profile page. #165

Added default birthday values to upcoming dates for Alice and Jeff in initial data #273

Able to save Interested in field now #163

Added more internationalisation (i18n) to profile html file #147

Able to save relationship status now #163

### CODE REVIEW

File: app/controllers/Profiles.java [hunk 1]
```diff
@@ -41,10 +41,12 @@ public static void updateInformation(String birthday, String relationshipStatus,
...
     DateFormat birthday_formatting = new SimpleDateFormat("MM/dd/yyyy");
-    try{
-      profile.birthday = (Date) birthday_formatting.parse(birthday);
-    } catch (java.text.ParseException e) {
-      e.printStackTrace();
+    if(birthday != null){
+      try{
+        profile.birthday = (Date) birthday_formatting.parse(birthday);
+      } catch (java.text.ParseException e) {
+        e.printStackTrace();
```
Comment (amshali) [inline app/controllers/Profiles.java]: you need to do something proper on the exception. It is very important to handle it well. We don't want to show the user the exception trace. Handle it, show a proper message to the user.

File: app/controllers/Profiles.java [hunk 2]
```diff
@@ -57,6 +59,8 @@ public static void updateInformation(String birthday, String relationshipStatus,
         e.printStackTrace();
       }
     }
+//    if(profile.relationshipStatus != null)
```
Comment (amshali) [inline app/controllers/Profiles.java]: remove the comments.

File: app/views/Users/profile.html [hunk 3]
```diff
@@ -121,11 +126,12 @@
             </tr>
           </table>
           <input type="submit" value="Save" id="updateInformation" />
+          <input type="reset" value="Cancel">
```
Comment (amshali) [inline app/views/Users/profile.html:135]: What will the Cancel do? Have you tested it?

Comment (fism88) [inline app/views/Users/profile.html:135]: Yes I tested it and it reverts back to the previous state for all the fields associated with it

**Comment by amshali** [comment]
What's up?

**Comment by fism88** [comment]
Does this now look better for the error messages?

**Comment by amshali** [comment]
No. Look how errors are handled here:
http://www.playframework.org/documentation/1.2.4/guide4

**Comment by amshali** [comment]
Also look at the signup page and how errors are handled there,

**Comment by fism88** [comment]
Right, the validation. I'm still working on the regex and testing because I am using validation.match to handle user input errors for calendar related fields.

**Comment by amshali** [comment]
As soon as you send an update I will review and merge it. Make sure you pull first.

**Comment by fism88** [comment]
Okay I moved the error messages to display all of them at the top of the left panel in red and tested to see if saving still works.

**Comment by fism88** [comment]
Just a heads up, I fixed some other things as well

**Comment by amshali** [comment]
let me test it

**Pull request merged**